### PR TITLE
Selenium driver for windows - Fixing session & cookie concent, and certificate parsing errors 

### PR DESCRIPTION
**Pull request by steinhaug**
### Background
Windows, selenium driver. When using latest release I got errors when loading URLs all the time, even wrong URLs were loaded when using Chrome as selenium browser. Also there were problems with lots of pages having cookie concent popping over the page, making the logic collapse as the cookie concent page were scraped by AutoGPT instead of the real page (after the concent which wasnt done).

### Changes
This change adds 2 configurable settings for the .env file, you can add a Chrome user-data-dir which is already logged in to all pages needed avoiding the Cookie concent problem. Also you can enable fixing the errors so many people are reporting on windows, and now it all works great.

### Documentation
I belive the changes should be selv explanatory, I did add some comments describing aswell.

### Test Plan
Should be tested on windows where you have problems with loading URL, there are loads of people reporting this.
Quick google search for the typical error: https://tinyurl.com/2jffs89j 

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

I havnt added tests, as I do not know how I add tests. :(

### CODE REVIEW

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Boostrix** [comment]
Note this looks like a useful change to be reviewed/integrated despite the ongoing re-arch effort, it's primarily config-level, with very few code changes. So definitely do consider for review/integration to help close some PRs. Thanks

File: .env.template [hunk 1]
```diff
@@ -165,6 +165,12 @@ OPENAI_API_KEY=your-openai-api-key
 ## BROWSE_SPACY_LANGUAGE_MODEL is used to split sentences. Install additional languages via pip, and set the model name here. Example Chinese:  python -m spacy download zh_core_web_sm
 # BROWSE_SPACY_LANGUAGE_MODEL=en_core_web_sm
 
+## WINDOWS CHROME SETTINGS - user-data-dir & errors
+## Lauch Google Chrome, pick a dir, example "C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" "--user-data-dir=G:\chrome-bank\selenium.driver"
+## Visit required sites, log into youtube, facebook and so. Make sure no pages ask for cookies opr other splash screens destroying the flow.
+#BROWSER_CHROME_USER_DATA_DIR="G:\\chrome-bank\\selenium.driver.python"
```
Comment (lc0rp) [inline .env.template:171]: Is this windows-specific? If so, I suggest prefixing with WIN or WINDOWS:
- WIN_BROWSER_CHROME_USER_DATA_DIR, or
- WIN_CHROME_USER_DATA_DIR

Comment (lc0rp) [reply .env.template:171]: I suggest dropping "WINDOWS" from line 168 if it's not windows specific.

File: autogpt/commands/web_selenium.py [hunk 2]
```diff
@@ -106,6 +106,13 @@ def scrape_text_with_selenium(url: str) -> tuple[WebDriver, str]:
             options.add_argument("--headless=new")
             options.add_argument("--disable-gpu")
 
+        if CFG.selenium_chrome_ignore_errors:
+            options.add_argument("--ignore-certificate-errors")
+            options.add_experimental_option('excludeSwitches', ['enable-logging'])
+
+        if CFG.selenium_chrome_user_data_dir != "":
```
Comment (lc0rp) [inline autogpt/commands/web_selenium.py:113]: Diito here, if windows specific, I'd prefix with "win" or "windows"

File: autogpt/config/config.py [hunk 3]
```diff
@@ -117,8 +120,7 @@ def __init__(self) -> None:
         # no user agent was found.
         self.user_agent = os.getenv(
             "USER_AGENT",
-            "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36"
-            " (KHTML, like Gecko) Chrome/83.0.4103.97 Safari/537.36",
+            "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36",
```
Comment (lc0rp) [inline autogpt/config/config.py:123]: This change may break things for people. If AutoGPT needs this to work on Windows, then I suggest we make it configurable, or set a different default based on sys.platform.

**Review by lc0rp — COMMENTED** [review]
Added some comments. This could go into 0.4.1 if addresed.

**Review by wisdoms6969 — APPROVED** [review]

**Review by wisdoms6969 — APPROVED** [review]

**Review by wisdoms6969 — APPROVED** [review]

**Review by wisdoms6969 — APPROVED** [review]

**Pull request closed without merging**